lara-typescript/src/section-authoring/api/lara-api-provider.ts

```diff
@@ -90,14 +90,27 @@ export const getLaraAuthoringAPI =
     return sendToLara({url: createPageUrl, method: "POST"});
   };
 
-  const updatePage: APIPageUpdateF = (args: {pageId: PageId, changes: Partial<IPage>}) => {
+  const updatePage: APIPageUpdateF = async (args: {pageId: PageId, changes: Partial<IPage>}) => {
     const {pageId, changes} = args;
-    const data = { id: pageId,
-                   page: { id: pageId,
-                           name: changes.name,
-                           isCompletion: changes.isCompletion,
-                           isHidden: changes.isHidden } };
-    return sendToLara({url: updatePageUrl(pageId), method: "PUT", body: data});
+    const pages = await getPages();
+    const indx = pages.findIndex(p => p.id === pageId);
+    if (indx > -1) {
+      if (indx < pages.length - 1 && changes.isCompletion) {
+        pages.push(pages.splice(indx, 1)[0]); // put the page at the end of the pages list
+      }
+      const newIndx = pages.length;
+      const nextPage = {... pages[newIndx], ...changes };
+      pages[newIndx] = nextPage;
```

Comment:
I'm not sure this matters, either, but might it be better to have the Rails side handle all of this?